test/components/bookmarksToolbarTest.js

```diff
@@ -0,0 +1,34 @@
+/* global describe, it, before */
+
+const Brave = require('../lib/brave')
+const messages = require('../../js/constants/messages')
+const {urlInput} = require('../lib/selectors')
+
+describe('bookmarksToolbar', function () {
+  function * setup (client) {
+    yield client
+      .waitUntilWindowLoaded()
+      .waitForVisible('#window')
+      .waitForVisible(urlInput)
+  }
+
+  describe('toolbar signal', function () {
+    Brave.beforeAll(this)
+    before(function *() {
+      yield setup(this.app.client)
+    })
+
+    it.skip('should open toolbar when signaled', function *() {
+      yield this.app.client
+        .ipcSend(messages.SHOW_BOOKMARKS_TOOLBAR)
+        .waitForExist('.bookmarksToolbar')
+    })
+
+    it.skip('should close toolbar when signaled', function *() {
+      yield this.app.client
+        .ipcSend(messages.SHOW_BOOKMARKS_TOOLBAR)
+        .ipcSend(messages.HIDE_BOOKMARKS_TOOLBAR)
+        .isExisting('.bookmarksToolbar').should.eventually.be.false
+    })
```

Comment:
Does this work? I don't think show/hide bookmarks is hooked up to anything so I can't imagine it would.   Previous test commits also didn't work when I ran the tests. I think you need to run `npm run watch-all` in one terminal and in the other do npm run test.

You can change a describe block to describe.only to run only that test or you can run `npm run test -- --grep=should\ close\ toolbar\ when\ signaled` for example.

Also I think that an IPC message is not correct for this, instead it should be a toggle of this pref: `bookmarks.toolbar.show`.